### ISSUE TITLE
Supplementary reads

### ISSUE THREAD

**Issue by kcleal** [ksahlin/StrobeAlign#13]
Hi,
Just a question - does strobealign support split-reads / supplementary mappings with a flag value of 2048? 
Thanks
Kez

**Comment by ksahlin** [comment]
Hi @kcleal,

Not yet, but I have found that it is possible to report supplementary alignments semi-easily with the current code structure. Release 0.7 will support supplementary alignments. Will keep this issue open until it is implemented. 

Release 0.6 will come in a couple of days with further improvements.

**Comment by kcleal** [comment]
Thanks!

Supplementary mappings are really handy for calling SVs, so would be great if you can implement this

**Comment by marcelm** [comment]
I realized just now that our tiny test genome (phiX) is circular and therefore good for testing supplementary alignments: Some of the phiX reads "wrap around" the FASTA reference, that is, a prefix aligns to the 3' end of the reference and a suffix aligns to its 5' end. This would need to be modelled with supplementary alignments.

**Comment by edg1983** [comment]
Supplementary reads are not yet implemented in the last release (0.7.1) right? I'm testing it to see how strobealign perform in our current WGS pipeline and having supp reads is crucial for our SV detection workflow.

**Comment by marcelm** [comment]
Yes, this is not implemented, but is definitely on the agenda.

**Comment by ksahlin** [comment]
Yes @edg1983, I was a bit hasty promising this in v0.7. Supplementary alignments is one of our highest priority features to implement though. So hopefully we will come back to this soon.

**Comment by kcleal** [comment]
Hi @kcleal, any ideas when this might be added? I would like to start using strobealign for SV analysis. Thanks!

**Comment by ksahlin** [comment]
Hi @kcleal,

Thanks for the interest! What I can say is that @marcelm is currently working on improving/speeding up extension alignment, and it is slightly related to supplementary alignments. Also, it is definitely in our interest to add such feature. But I don't wan't to interfere with his todo list, so I will simply pass this question to @marcelm.


To Marcel: I guess we also need to assess if it fits the number of hours left on the project and the priorities.

**Comment by prototaxites** [comment]
Hi @ksahlin, I am also interested in a feature like this in Strobealign (in our case, for mapping Hi-C reads) - is this still somewhere on the development roadmap?

**Comment by ksahlin** [comment]
It is definitely on the roadmap, we have just been prioritising other things. The recent [multi-context seed update](https://www.biorxiv.org/content/10.1101/2024.10.29.620855v1) to strobealign enables better split read detection (suppl. mappings). Now we "just" have to implement the actual functionality. I will discuss with @marcelm and @Itolstoganov .
